File: src/main/java/it/contrader/dto/HardwareDTO.java
package it.contrader.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class HardwareDTO {
	private Integer idHardware;
	private String name;
	private String description;
	private ThingDTO thing = new ThingDTO();
}

File: IncAng/src/dto/HardwareDTO.ts
import { ThingDTO } from './ThingDTO';

export class HardwareDTO{
    idHardware: number;
    name: String;
    description: String;
    thing: ThingDTO;


constructor(id: number, name: String, description: String, thingDTO: ThingDTO ){
    this.idHardware=id;
    this.name= name;
    this.description=description;
    this.thing = thingDTO;
}
}